web/src/components/MemoEditor/Editor/index.tsx

```diff
@@ -127,15 +127,16 @@ const Editor = forwardRef(function Editor(props: Props, ref: React.ForwardedRef<
 
   return (
     <div className={"common-editor-wrapper " + className}>
-      <textarea
-        className="common-editor-inputer"
-        rows={1}
-        placeholder={placeholder}
-        ref={editorRef}
-        onPaste={onPaste}
-        onInput={handleEditorInput}
-      ></textarea>
-      <TagSuggestions editorRef={editorRef} editorActions={ref} />
+      <TagSuggestions editorRef={editorRef} editorActions={ref}>
```

Comment:
Do we really need to put `textarea` as a child into `TagSuggestions`? Maybe we can use global event detector `window.addEventListener` in `TagSuggestions`.

Reply:
I will try to revert into a sibling of textarea and use `ref.addEventListener`. However, there is one callback closure problem there to solve.

<img width="1192" alt="closure-problem" src="https://github.com/usememos/memos/assets/5403694/9d49ccd9-7d91-4e41-8b82-27d62696d08b">

In this fragment of the previous solution, a console log will always print an empty array (expected would be the first time empty but then populated).

Event handler callback remembers variables in a closure at the moment it's registered. In that closure, `tags` are an empty array.

I will try a solution with`useRef`. Last time I ended up writing `useRef` everywhere, but I will give it one more try.